microceph/ceph/replication.go

```diff
@@ -57,7 +60,9 @@ func GetReplicationStateMachine(initialState ReplicationState) *stateless.StateM
 		Permit(constants.EventEnableReplication, StateEnabledReplication).
 		OnEntryFrom(constants.EventDisableReplication, disableHandler).
 		InternalTransition(constants.EventListReplication, listHandler).
-		InternalTransition(constants.EventDisableReplication, disableHandler)
+		InternalTransition(constants.EventDisableReplication, disableHandler).
+		InternalTransition(constants.EventPromoteReplication, promoteHandler).
+		InternalTransition(constants.EventDemoteReplication, demoteHandler)
```

Comment:
Hm, not sure I understand -- why allow promote/demote when replication is disabled?

Reply:
promote/demote/list are workload level operations (they do not operate on a single resource) hence it cannot be dependant on the state of any single pool or image and should be operable on both states.